pkg/decision/evaluator/matchers/semver.go

```diff
@@ -65,16 +66,12 @@ func (sv SemanticVersion) compareVersion(attribute string) (int, error) {
 			return -1, nil
 		case !sv.isNumber(versionParts[idx]):
 			// Compare strings
-			if versionParts[idx] < targetedVersionParts[idx] {
-				return -1, nil
-			} else if versionParts[idx] > targetedVersionParts[idx] {
-				return 1, nil
+			if result := sv.compareVersionStrings(versionParts[idx], targetedVersionParts[idx], attribute); result != 0 {
```

Comment:
any reason to put it in a separate function?
if we need to, please rename this function name, compareBuildMetadata.

Reply:
had to create these functions to resolve cyclo-complexity warning thrown by linter.